### PR TITLE
HTS-955: Remove Global objects and use dependency injection instead

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/config/ErrorHandler.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import play.api.mvc.Request
+import play.twirl.api.Html
+import uk.gov.hmrc.helptosavefrontend.models.HtsContext
+import uk.gov.hmrc.helptosavefrontend.views
+import uk.gov.hmrc.play.bootstrap.http.FrontendErrorHandler
+
+@Singleton
+class ErrorHandler @Inject() (implicit val messagesApi: MessagesApi, val appConfig: FrontendAppConfig) extends FrontendErrorHandler {
+  override def standardErrorTemplate(pageTitle: String, heading: String, message: String)(implicit request: Request[_]): Html = {
+    implicit val htsContext: HtsContext = HtsContext(true)
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/config/ErrorHandler.scala]: this hacky bit is inherited from existing code `implicit val htsContext: HtsContext = HtsContext(authorised = false)` from the removed `FrontendGloal` class except that i used `true` flag above

File: app/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnector.scala [hunk 2]
```diff
@@ -20,18 +20,17 @@ import cats.data.EitherT
...
+import uk.gov.hmrc.helptosavefrontend.config.{FrontendAppConfig, WSHttp}
 import uk.gov.hmrc.helptosavefrontend.connectors.HelpToSaveConnectorImpl.{ECResponseHolder, GetEmailResponse}
-import uk.gov.hmrc.helptosavefrontend.connectors.HelpToSaveConnectorImpl.URLS._
 import uk.gov.hmrc.helptosavefrontend.models._
 import uk.gov.hmrc.helptosavefrontend.models.eligibility.{EligibilityCheckResponse, EligibilityCheckResult}
 import uk.gov.hmrc.helptosavefrontend.models.userinfo.MissingUserInfo
 import uk.gov.hmrc.helptosavefrontend.util.HttpResponseOps._
 import uk.gov.hmrc.helptosavefrontend.util.{Email, Result, base64Encode, maskNino}
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnector.scala]: we shouldn't need to import the global execution context - we should pass it in implicitly in the calls where it's needed

File: app/uk/gov/hmrc/helptosavefrontend/audit/HTSAuditor.scala [hunk 3]
```diff
@@ -18,18 +18,16 @@ package uk.gov.hmrc.helptosavefrontend.audit
...
 import javax.inject.{Inject, Singleton}
 
-import uk.gov.hmrc.helptosavefrontend.config.FrontendAuditConnector
 import uk.gov.hmrc.helptosavefrontend.models.HTSEvent
-import uk.gov.hmrc.helptosavefrontend.util.{Logging, NINO, NINOLogMessageTransformer}
 import uk.gov.hmrc.helptosavefrontend.util.Logging._
+import uk.gov.hmrc.helptosavefrontend.util.{Logging, NINO, NINOLogMessageTransformer}
 import uk.gov.hmrc.play.audit.http.connector.AuditConnector
 
 import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/audit/HTSAuditor.scala]: we shouldn't need the global execution context here

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**